src/pkgcheck/checks/metadata.py

```diff
@@ -586,6 +587,68 @@ def feed(self, pkgs):
             yield MissingLocalUseDesc(sorted(missing_desc), pkg=pkg)
 
 
+class UseFlagWithoutDeps(results.VersionResult, results.Warning):
+    """Special Use flag without effect on dependencies and provide little
+    utility.
+
+    Various USE flags, such as "ipv6", should be always turned on or off, and
+    their existence is questionable, in cases were it doesn't introduce new
+    dependencies. Other USE flags, such as "bash-completion", without any new
+    dependencies, are probable violators of small files QA policy [#]_.
+
+    In cases where this USE flag is appropriate, you can ignore this warning by
+    adding a description to this USE flag in ``metadata.xml`` file and thus
+    making it a local USE flag instead of global one.
+
+    .. [#] https://projects.gentoo.org/qa/policy-guide/installed-files.html#pg0301
+    """
+
+    def __init__(self, flags, **kwargs):
+        super().__init__(**kwargs)
+        self.flags = tuple(flags)
+
+    @property
+    def desc(self):
+        s = pluralism(self.flags)
+        flags = ', '.join(self.flags)
+        return f'special small-files USE flag{s} without affect on dependencies: [ {flags} ]'
+
+
+class UseFlagsWithoutEffectsCheck(GentooRepoCheck):
+    """Check for USE flags without effects."""
+
+    known_results = frozenset({
+        UseFlagWithoutDeps,
+    })
+
+    warn_use_small_files = frozenset({
+        'ipv6', 'unicode',
+        'bash-completion', 'zsh-completion', 'vim-syntax',
```

Comment:
Let's add in fish-completion too, logrotate, maybe USE=systemd (this may have more FPs we must look at, so put a TODO instead?)